[2025-09-18T06:11:25.978+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: batch_ingestion_dag.extract_mysql manual__2025-09-18T06:11:22.647336+00:00 [queued]>
[2025-09-18T06:11:25.987+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: batch_ingestion_dag.extract_mysql manual__2025-09-18T06:11:22.647336+00:00 [queued]>
[2025-09-18T06:11:25.988+0000] {taskinstance.py:2171} INFO - Starting attempt 1 of 2
[2025-09-18T06:11:26.000+0000] {taskinstance.py:2192} INFO - Executing <Task(PythonOperator): extract_mysql> on 2025-09-18 06:11:22.647336+00:00
[2025-09-18T06:11:26.008+0000] {standard_task_runner.py:60} INFO - Started process 202 to run task
[2025-09-18T06:11:26.011+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'batch_ingestion_dag', 'extract_mysql', 'manual__2025-09-18T06:11:22.647336+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/batch_ingestion_dag.py', '--cfg-path', '/tmp/tmp_flb_h44']
[2025-09-18T06:11:26.015+0000] {standard_task_runner.py:88} INFO - Job 2: Subtask extract_mysql
[2025-09-18T06:11:26.071+0000] {task_command.py:423} INFO - Running <TaskInstance: batch_ingestion_dag.extract_mysql manual__2025-09-18T06:11:22.647336+00:00 [running]> on host 1f9d8fdca5a4
[2025-09-18T06:11:26.150+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='batch_ingestion_dag' AIRFLOW_CTX_TASK_ID='extract_mysql' AIRFLOW_CTX_EXECUTION_DATE='2025-09-18T06:11:22.647336+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-09-18T06:11:22.647336+00:00'
[2025-09-18T06:11:26.151+0000] {batch_ingestion_dag.py:28} INFO - Starting extraction from MySQL...
[2025-09-18T06:11:26.152+0000] {crypto.py:82} WARNING - empty cryptography key - values will not be stored encrypted.
[2025-09-18T06:11:26.153+0000] {base.py:83} INFO - Using connection ID '***_db' for task execution.
[2025-09-18T06:11:26.257+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.9/site-packages/***/providers/common/sql/hooks/sql.py:215 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2025-09-18T06:11:26.290+0000] {batch_ingestion_dag.py:33} INFO - Extracted 5 records from MySQL and saved to /tmp/orders.csv.
[2025-09-18T06:11:26.303+0000] {xcom.py:664} ERROR - Object of type Timestamp is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2025-09-18T06:11:26.304+0000] {batch_ingestion_dag.py:41} ERROR - Failed to extract data from MySQL: Object of type Timestamp is not JSON serializable
[2025-09-18T06:11:26.305+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/serialization/serde.py", line 180, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'pandas._libs.tslibs.timestamps.Timestamp'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/batch_ingestion_dag.py", line 37, in extract_data_from_mysql
    kwargs["ti"].xcom_push(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 2981, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/xcom.py", line 247, in set
    value = cls.serialize_value(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/xcom.py", line 662, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/lib/python3.9/json/__init__.py", line 234, in dumps
    return cls(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/local/lib/python3.9/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.9/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/local/lib/python3.9/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Timestamp is not JSON serializable
[2025-09-18T06:11:26.317+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=batch_ingestion_dag, task_id=extract_mysql, execution_date=20250918T061122, start_date=20250918T061125, end_date=20250918T061126
[2025-09-18T06:11:26.328+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 2 for task extract_mysql (Object of type Timestamp is not JSON serializable; 202)
[2025-09-18T06:11:26.345+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-09-18T06:11:26.368+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
